### PR TITLE
Slow down sync when token is required

### PR DESCRIPTION
**Pull request by fao89**
https://pulp.plan.io/issues/7917
closes #7917

### CODE REVIEW

File: CHANGES/7917.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Slow down sync when token is required
```
Comment (bmbouter) [inline CHANGES/7917.bugfix:1]: Can we reword this to focus on what it's fixing instead? Perhaps `Avoid rate limiting by slowing down sync when token is required`.

**Comment by bmbouter** [comment]
Also will we be putting this on master too? If so, it probably should go there first and then be brought back as a backport.

**Comment by fao89** [comment]
> Also will we be putting this on master too? If so, it probably should go there first and then be brought back as a backport.

I was thinking: hardcoded on 0.5 and using a setting on master

**Comment by bmbouter** [comment]
If we're going to make a "slow down" feature we should implement leaky bucket and have the requests per second rate be on the Remote. It would be much more usable for example if we know the remote site has 12 req / sec then someone could set it to 11 and know it's the right value.

**Comment by fao89** [comment]
I agree, I'll change this PR to the master branch then

File: pulp_ansible/app/downloaders.py [hunk 2]
```diff
@@ -128,6 +128,7 @@ async def _run_with_additional_headers(self, headers):
             DownloadResult: Contains information about the result. See the DownloadResult docs for
                  more information.
         """
+        await asyncio.sleep(0.2)
```
Comment (fao89) [inline pulp_ansible/app/downloaders.py:131]: @newswangerd do you think 0.2 is good?
ref: https://hackmd.io/@fao89/rate_limiting

**Review by chouseknecht — APPROVED** [review]

**Review by bmbouter — APPROVED** [review]
Thank you!

File: pulp_ansible/app/downloaders.py (hunk 2 above)
Comment (newswangerd) [reply pulp_ansible/app/downloaders.py:131]: Sorry I'm late to the party. I would have suggested something more like 0.5. The rationale being that we know the limit is 15 requests per second and concurrent connections defaults to 10, so that would give us a theoretical upper bound of 20 requests per second, assuming all requests are instant (which they aren't). That should roughly guarantee that nobody runs into this issue.

0.2 is probably fine though.
